pkg/natdiscovery/request_send.go

```diff
@@ -0,0 +1,152 @@
+/*
+© 2021 Red Hat, Inc. and others
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package natdiscovery
+
+import (
+	"net"
+
+	"github.com/google/gopacket/routing"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+	"k8s.io/klog"
+
+	natproto "github.com/submariner-io/submariner/pkg/natdiscovery/proto"
+)
+
+func (nd *natDiscovery) sendCheckRequestByRemoteID(id string) error {
+	remoteNAT := nd.remoteEndpoints[id]
+	return nd.sendCheckRequest(remoteNAT)
+}
+
+func (nd *natDiscovery) sendCheckRequest(remoteNAT *remoteEndpointNAT) error {
+	var errPrivate, errPublic error
+	var reqID uint64
+	if remoteNAT.Endpoint.Spec.PrivateIP != "" {
+		reqID, errPrivate = nd.sendCheckRequestToTargetIP(remoteNAT, remoteNAT.Endpoint.Spec.PrivateIP)
+		if errPrivate == nil {
+			remoteNAT.lastPrivateIPRequestID = reqID
+		}
+	}
+
+	if remoteNAT.Endpoint.Spec.PublicIP != "" {
+		reqID, errPublic = nd.sendCheckRequestToTargetIP(remoteNAT, remoteNAT.Endpoint.Spec.PublicIP)
+		if errPublic == nil {
+			remoteNAT.lastPublicIPRequestID = reqID
+		}
+	}
+
+	if errPrivate != nil && errPublic != nil {
+		return errors.Errorf("error happened trying to discover both public & private IPs of endpoint %q, [%s, %s]",
```

Comment:
"error while trying to discover both public ....."

Reply:
I forgot about this one, will handle on next interation,

Reply:
Done.